bn_fast_s_mp_sqr.c

```diff
@@ -66,26 +66,26 @@ int fast_s_mp_sqr(const mp_int *a, mp_int *b)
        * we halve the distance since they approach at a rate of 2x
        * and we have to round because odd cases need to be executed
        */
-      iy = MIN(iy, ((ty-tx)+1)>>1);
+      iy = MIN(iy, ((ty-tx)+1)/2);
```

Comment:
I'm very unsure if I should like or dislike the "refactor without shift" (currently 60fb7e4c56d81dbce0b2193440c4a170bc8f62da) commit

Reply:
Just to be clear: I like the idea behind the change!

and this is just a wild guess: can't this fire back for some older/special compilers?

Reply:
The [MISRA](https://en.wikipedia.org/wiki/MISRA_C) rules allow bitwise operations (including shift) only with `unsigned` integers, but not with `signed`.

When the code looks like:
```
  /* now divide in two */
   B = B >> 1;
```
the use of shift is a way of obfuscate the source.

Some files (`bn_mp_exptmod_fast.c`, `bn_s_mp_exptmod.c`) made an intensive use of bitwise operations on `signed`, but I have not dared to refactor them.